src/main/ext/nix_env.cljs

```diff
@@ -38,20 +39,24 @@
                             (catch js/Error _ nil))]))))
        (filter not-empty)))
 
-(defn get-nix-env-sync [options]
-  (-> (get-shell-env-cmd options)
-      (execSync (clj->js {:cwd (dirname (:nix-config options))}))
-      (.toString)
-      (parse-exported-vars)))
-
-(defn get-nix-env-async [options]
-  (let [env-result (p/deferred)]
-    (exec (get-shell-env-cmd options)
-          (clj->js {:cwd (dirname (:nix-config options))})
-          (fn [err result]
+(defn get-nix-env-sync [options log-channel]
+  (let [cmd (get-shell-env-cmd options)]
+    (w/write-log log-channel (str "Running command synchronously: " cmd))
+    (-> (execSync (clj->js cmd {:cwd (dirname (:nix-config options))}))
+        (.toString)
+        (parse-exported-vars))))
+
+(defn get-nix-env-async [options log-channel]
+  (let [env-result (p/deferred)
+        cmd (get-shell-env-cmd options)]
+    (w/write-log log-channel (str "Running command asynchronously: " cmd))
+    (exec (clj->js cmd {:cwd (dirname (:nix-config options))})
+          (fn [err result stderr]
             (if (nil? err)
               (p/resolve! env-result result)
-              (p/reject! env-result err))))
+              (
```

Comment:
I think this is a "syntax" error:

```suggestion
              (do
```

Without this fix, I get:

```
2023-09-14 11:29:31.332 [error] TypeError: Cannot read properties of undefined (reading 'cljs$core$IFn$_invoke$arity$1')
	at /root/.vscode-server/extensions/arrterian.nix-env-selector-1.0.9/dist/main.js:2608:261
	at ChildProcess.exithandler (node:child_process:408:5)
	at ChildProcess.emit (node:events:513:28)
	at maybeClose (node:internal/child_process:1093:16)
	at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

The generated JS code is:

```
        g=vscode.window.write_log(b,["Error applying environment: ",cljs.core.str.cljs$core$IFn$_invoke$arity$1(g)].join(""));
        return g.cljs$core$IFn$_invoke$arity$1 ? g.cljs$core$IFn$_invoke$arity$1(e) : g.call(null,e)});
```

which is 

So that's assigning `g = vscode.window.write_log(...)` and then accessing `g.something`, but `write_log()` probably returns nothing (`undefined`).

I feel like any type-checker would have saved quite some debugging time here.